topydo/lib/PrettyPrinterFilter.py

```diff
@@ -57,15 +60,21 @@ def filter(self, p_todo_str, p_todo):
             except KeyError:
                 pass
 
-            p_todo_str = color + p_todo_str + NEUTRAL_COLOR
-
+            p_todo_str = '%s%s%s' % (color, p_todo_str, NEUTRAL_COLOR)
             if config().highlight_projects_contexts():
                 p_todo_str = re.sub(
                     r'\B(\+|@)(\S*\w)',
-                    PROJECT_COLOR + r'\g<0>' + color,
+                    #PROJECT_COLOR + r'\g<0>' + color,
+                    lambda m: (
+                        CONTEXT_COLOR if m.group(0)[0] == "+"
+                        else PROJECT_COLOR)+m.group(0)+color,
                     p_todo_str)
-
-            p_todo_str += NEUTRAL_COLOR
+            p_todo_str = re.sub(r'\w(\S*:[^/\s]\S*)\w',METADATA_COLOR+r'\g<0>'+color,p_todo_str)
+            # add LINK_COLOR to anything like :  (one group od charater and a :)(// and non whitespace chars)
+            p_todo_str = re.sub(r'\s(\w*:){1}(//\S*\w)',' '+LINK_COLOR+r'\1\2'+color,p_todo_str)
```

Comment:
Thanks for the Pythex links. I would suggest the following:

`(^|\s)(\w+:){1}(//\S+)` ([link](http://pythex.org/?regex=%28^|\s%29%28\w%2B%3A%29{1}%28%2F%2F\S%2B%29&test_string=http%3A%2F%2Fexample.org%20http%3A%2F%2Fexample.org%20Lorem%20Ipsum%20Foo%20url%3Ahttp%3A%2F%2Fexample.org%20http%3A%2F%2Fexample.org%20%3A%2F%2Fexample.org&ignorecase=0&multiline=0&dotall=0&verbose=0))

Because your expression would match `://example.org`.